pkg/relayer/proxy/relay_verifier.go

```diff
@@ -18,81 +14,21 @@ func (rp *relayerProxy) VerifyRelayRequest(
-			"application_address": relayRequest.Meta.SessionHeader.ApplicationAddress,
-			"service_id":          relayRequest.Meta.SessionHeader.Service.Id,
-		}).
-		Msg("verifying relay request signature")
-
-	// extract the relay request's ring signature
-	if relayRequest.Meta == nil {
-		return ErrRelayerProxyEmptyRelayRequestSignature.Wrapf(
-			"request payload: %s", relayRequest.Payload,
-		)
-	}
-	signature := relayRequest.Meta.Signature
-	if signature == nil {
-		return sdkerrors.Wrapf(
-			ErrRelayerProxyInvalidRelayRequest,
-			"missing signature from relay request: %v", relayRequest,
-		)
-	}
-
-	ringSig := new(ring.RingSig)
-	if err := ringSig.Deserialize(ring_secp256k1.NewCurve(), signature); err != nil {
-		return sdkerrors.Wrapf(
-			ErrRelayerProxyInvalidRelayRequestSignature,
-			"error deserializing ring signature: %v", err,
-		)
-	}
-
-	if relayRequest.Meta.SessionHeader.ApplicationAddress == "" {
-		return sdkerrors.Wrap(
-			ErrRelayerProxyInvalidRelayRequest,
-			"missing application address from relay request",
-		)
-	}
-
-	// get the ring for the application address of the relay request
-	appAddress := relayRequest.Meta.SessionHeader.ApplicationAddress
-	appRing, err := rp.ringCache.GetRingForAddress(ctx, appAddress)
-	if err != nil {
-		return sdkerrors.Wrapf(
-			ErrRelayerProxyInvalidRelayRequest,
-			"error getting ring for application address %s: %v", appAddress, err,
-		)
-	}
-
-	// verify the ring signature against the ring
-	if !ringSig.Ring().Equals(appRing) {
-		return sdkerrors.Wrapf(
-			ErrRelayerProxyInvalidRelayRequestSignature,
-			"ring signature does not match ring for application address %s", appAddress,
-		)
-	}
-
-	// get and hash the signable bytes of the relay request
-	requestSignableBz, err := relayRequest.GetSignableBytesHash()
-	if err != nil {
-		return ErrRelayerProxyInvalidRelayRequest.Wrapf("error getting signable bytes: %v", err)
+	if err := rp.ringCache.VerifyRelayRequestSignature(ctx, relayRequest); err != nil {
+		return err
 	}
 
-	// verify the relay request's signature
-	if valid := ringSig.Verify(requestSignableBz); !valid {
-		return sdkerrors.Wrapf(
-			ErrRelayerProxyInvalidRelayRequestSignature,
-			"invalid ring signature",
-		)
-	}
+	// Application address is used to verify the relayRequest signature, it is
+	// guaranteed to be present in the relayRequest since the signature has already
+	// been verified.
+	appAddress := relayRequest.GetMeta().GetSessionHeader().GetApplicationAddress()
 
 	// Query for the current session to check if relayRequest sessionId matches the current session.
 	rp.logger.Debug().
 		Fields(map[string]any{
-			"session_id":          relayRequest.Meta.SessionHeader.SessionId,
-			"application_address": relayRequest.Meta.SessionHeader.ApplicationAddress,
-			"service_id":          relayRequest.Meta.SessionHeader.Service.Id,
+			"session_id":          relayRequest.GetMeta().GetSessionHeader().GetSessionId(),
```

Comment:
Let's assign sessionHeader to a local variable and simplify this and the other usage below.